iceoryx_posh/test/integrationtests/test_client_server.cpp

```diff
@@ -450,4 +482,121 @@ TEST_F(ClientServer_test, ClientTakesResponseUnblocksServerSendingResponse)
     EXPECT_THAT(wasResponseSent.load(), Eq(true));
 }
 
+#ifdef TEST_WITH_HUGE_PAYLOAD
+
+TEST_F(BigPayloadClientServer_test, TypedApiWithBigPayloadWithMatchingOptionsWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "9838d2dc-bd87-42aa-b581-a9526e35e46a");
+
+    constexpr int64_t SEQUENCE_ID{73};
+    constexpr uint64_t FIRST{4095};
+    constexpr uint64_t LAST{BIG_PAYLOAD_SIZE - 1};
+    constexpr uint64_t STEP{4096};
+    constexpr uint8_t SHIFT{13U};
+
+    Client<BigPayloadStruct, BigPayloadStruct> client{sd};
+    Server<BigPayloadStruct, BigPayloadStruct> server{sd};
+
+    // send request
+    {
+        auto loanResult = client.loan();
+        ASSERT_FALSE(loanResult.has_error());
+        auto& request = loanResult.value();
+        request.getRequestHeader().setSequenceId(SEQUENCE_ID);
+        for (uint64_t i = FIRST; i <= LAST; i += STEP)
+        {
+            request->bigPayload[i] = static_cast<uint8_t>((i / STEP) % 256U);
```

Comment:
(optional) To give more context understanding I would use here `std::numeric_limits<uint8_t>::max()`. Maybe also add a little type alias to `BigPayloadStruct` and use it here.

Reply:
`std::numeric_limits<uint8_t>::max().` won't work but yeah, a constexpr for this magic number would be nice. But on the other hand, I think this is not even necessary since the cast to uint8_t does already what wants to be achieved by `%256`.

Reply:
That's what I wanted to say. 
This modulo is unnecessary here.

I am not sure what would you expect the alias to be.
`uint8_t` is precisely what I needed here and what I am used to use when I refer to 'just some bytes in memory'.

Reply:
And I can get rid of the cast.
Good idea